### PR TITLE
Support async/ sync for subscription instance creation when creating a new subscription

### PR DESCRIPTION
**Pull request by TronTram**
# Motivation / Background
- User story: https://thepressingly.atlassian.net/browse/PINET-484
- Technical design: https://docs.google.com/document/d/1S6e6DWKXaPXTZZsKGgyawpuoPY-p1PPv_p3XLIFcXqo/edit#bookmark=id.xzjv44cvz0p5

*This Pull Request has been created because:*
- To provide flexibility in how subscription instances are created, accommodating both immediate and deferred processing needs.
- The default behavior is set to false, meaning the subscription instance will be created immediately after the subscription is created, unless specified otherwise.

# Detail

*This Pull Request changes:*

List of things you do in this PR
- Extend Subscriptions::CreateService to accept an `async`, allowing callers to specify if the processing should be synchronous or asynchronous. The default is false -> create a subscription instance immediately after create a subscription.

# Additional information

*TIP: Provide additional information such as screenshots, benchmarks, reference to other repositories or alternative solutions*

# Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Comment by tnlong311** [comment]
The title seems to not relevant to this PR description. Can you please update the title?

File: app/services/subscriptions/create_service.rb [hunk 1]
```diff
@@ -277,5 +283,10 @@ def billable_subscriptions(new_subscription)
 
       billable_subscriptions
     end
+
+    def process_subscription_instance_creation(subscription)
```
Comment (tnlong311) [inline app/services/subscriptions/create_service.rb:287]: why not `create_subscription_instance` for simple name?

File: spec/requests/api/v1/subscriptions_controller_spec.rb [hunk 2]
```diff
@@ -36,22 +36,19 @@
       }
     end
 
-    it 'creates a subscription instance and enqueues a job to create a subscription instance' do
+    it 'creates a subscription and then create a subscription instance' do
```
Comment (tnlong311) [inline spec/requests/api/v1/subscriptions_controller_spec.rb:39]: typo "and then *creates"

**Review by ghun131 — APPROVED** [review]
lgtm!

**Pull request merged**